File: iMMPI/iMMPI/AnalyserTableViewCell.m
//
//  AnalyserTableViewCell.m
//  iMMPI
//
//  Created by Egor Chiglintsev on 30.12.12.
//  Copyright (c) 2012 Egor Chiglintsev. All rights reserved.
//

#if (!__has_feature(objc_arc))
#error "This file should be compiled with ARC support"
#endif

#import "AnalyserTableViewCell.h"


#pragma mark -
#pragma mark AnalyserTableViewCell private

@interface AnalyserTableViewCell()
{
    IBOutlet UILabel *_groupNameLabel;
    IBOutlet UILabel *_scoreLabel;
    
    IBOutlet NSLayoutConstraint *_lcGroupNameLabelLeft;
}

@end


#pragma mark -
#pragma mark AnalyzerTableViewCell implementation

@implementation AnalyserTableViewCell

#pragma mark -
#pragma mark properties

- (void) setGroupNameOffset: (CGFloat) groupNameOffset
{
    _groupNameOffset = groupNameOffset;
    _lcGroupNameLabelLeft.constant = _groupNameOffset + 20.0;
    [self setNeedsUpdateConstraints];
}

@end

File: iMMPI/iMMPI/AnalyserTableViewCell.swift
import UIKit

final class AnalyserTableViewCell: UITableViewCell, AnalyserCellProtocol {
    @objc @IBOutlet fileprivate(set) var titleLabel: UILabel?
    @objc @IBOutlet fileprivate(set) var scoreLabel: UILabel?
    @objc @IBOutlet fileprivate(set) var indexLabel: UILabel?

    var titleOffset: CGFloat = 0 {
        didSet {
            setNeedsUpdateConstraints()
        }
    }

    @objc @IBOutlet fileprivate var titleLabelLeft: NSLayoutConstraint?
}


extension AnalyserTableViewCell {
    override func updateConstraints() {
        titleLabelLeft?.constant = titleOffset + 20.0
        super.updateConstraints()
    }
}
